src/Domain/Insights/InsightFactory.php

```diff
@@ -118,15 +118,14 @@ private function runInsightCollector(OutputInterface $consoleOutput): void
     private function loadInsights(array $insights): array
     {
         $insightsAdded = [];
-        $cwd = getcwd() ?? $this->config->getDirectories();
 
         foreach ($insights as $insight) {
             /** @var InsightLoader $loader */
             foreach ($this->insightLoaders as $loader) {
                 if ($loader->support($insight)) {
                     $insightsAdded[] = $loader->load(
                         $insight,
-                        (string) $cwd,
+                        (string) (getcwd() ?? $this->config->getCommonPath()),
```

Comment:
Can we keep this outside the double foreach ? It should not change on each loop.

Reply:
Yes, Thanks! Done.